addons/l10n_th/views/report_invoice.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH'" >
+                <span t-field="o.l10n_th_company_info" />
+                <span t-field="o.l10n_th_branch_number" />
+            </t>
+        </xpath>
+        <xpath expr="//h2" position="before">
+            <t t-set="use_custom_l10n_header" t-value="o.l10n_th_custom_header and o.company_id.account_fiscal_country_id.code == 'TH'"/>
+            <h2 t-if="use_custom_l10n_header">
+                <span t-if="o.debit_origin_id">Debit Note</span>
+                <span t-if="not o.debit_origin_id">Tax Invoice</span>
+                <span t-if="o.name != '/'" t-field="o.name"/>
+            </h2>
+        </xpath>
+        <xpath expr="//h2[2]" position="attributes">
+            <attribute name="t-if">not use_custom_l10n_header</attribute>
+        </xpath>
+        <xpath expr="//t[@t-call='account.document_tax_totals']" position="replace">
+            <t t-if="o.l10n_th_custom_header and o.company_id.account_fiscal_country_id.code == 'TH' and o.debit_origin_id">
+                <t t-call="l10n_th.document_tax_totals"/>
+            </t>
+            <t t-else="">
+                <t t-call="account.document_tax_totals"/>
+            </t>
+        </xpath>
+    </template>
+
+    <template id="document_tax_totals">
+        <tr class="border-black o_subtotal">
+            <td><strong>Original Invoiced Amount (1)</strong></td>
+            <td class="text-end">
+                <span
+                    t-att-class="oe_subtotal_footer_separator"
+                    t-field="o.debit_origin_id.amount_untaxed"
+                />
+            </td>
+        </tr>
+        <tr>
+            <td><span class="text-nowrap">Invoice Amount (2)</span></td>
+            <td class="text-end o_price_total">
+                <span class="text-nowrap" t-field="o.amount_untaxed" />
+            </td>
+        </tr>
+        <t t-set="amount_difference" t-value="o.amount_untaxed - o.debit_origin_id.amount_untaxed"/>
```

Comment:
that is weird, because the debit note normally contains the amount you add, so what is the point in calculating the differences between the debit note amount and the original invoice amount?

Reply:
It is a legal requirement from TH government
the detail can be found on this task https://www.odoo.com/web#id=2879718&cids=5&menu_id=4720&action=333&active_id=1822&model=project.task&view_type=form point 1.3.2